File: administration/src/util/sentry.ts
import * as Sentry from '@sentry/react'

import { isProductionEnvironment } from './helper'

export const initSentry = (): void => {
  Sentry.init({
    dsn: 'https://c560d6febb909858b17b92f91d418183@sentry.tuerantuer.org/6',
    release: VITE_BUILD_VERSION_NAME,
    integrations: [Sentry.browserTracingIntegration()],
    // https://docs.sentry.io/concepts/key-terms/tracing/
    tracesSampleRate: 0.05,
  })
}

export const reportErrorToSentry = (error: unknown): void => {
  if (isProductionEnvironment()) {
    Sentry.captureException(error)
  } else {
    console.error(error)
  }
}

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/shared/utils/Sentry.kt
package app.ehrenamtskarte.backend.shared.utils

import app.ehrenamtskarte.backend.BuildConfig
import io.sentry.Sentry

fun initializeSentry() {
    Sentry.init { options ->
        options.dsn = "https://c0c177b8fa35e0fd90d788b0d02bcb2c@sentry.tuerantuer.org//7"
        options.release = BuildConfig.VERSION_NAME
        // https://docs.sentry.io/concepts/key-terms/tracing/
        options.tracesSampleRate = 0.05
    }
}

fun reportErrorToSentry(e: Exception) {
    if (Sentry.isEnabled()) {
        Sentry.captureException(e)
    }
}
